onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -108,8 +135,13 @@ std::unique_ptr<KernelRegistry> RegisterKernels() {
       KERNEL_CREATE_INFO_VERSIONED(10, 10, MaxPool, kMSInternalNHWCDomain),
       KERNEL_CREATE_INFO_VERSIONED(11, 11, MaxPool, kMSInternalNHWCDomain),
       KERNEL_CREATE_INFO(12, MaxPool, kMSInternalNHWCDomain),
+      // because the F16 macro may be expanded to empty, so don't add ","
+      KERNEL_CREATE_INFO_VERSIONED_FP16(8, 9, MaxPool, kMSInternalNHWCDomain)
+          KERNEL_CREATE_INFO_VERSIONED_FP16(10, 10, MaxPool, kMSInternalNHWCDomain)
+              KERNEL_CREATE_INFO_VERSIONED_FP16(11, 11, MaxPool, kMSInternalNHWCDomain)
+                  KERNEL_CREATE_INFO_FP16(12, MaxPool, kMSInternalNHWCDomain)
```

Comment:
given this isn't going to scale well as we add more fp16 kernels, should we just use the regular macros and put all the fp16 registrations inside one #ifdef at the end?

Reply:
Personally, I think that we put all fp16 registrations in one #ifdef is less readable than putting all types of the kernel together, for example, we can easily find which op hasn't supported FP16. And we only need to change the op name .
Since you don't like this style, I can change it back to normal way.